syntax = "proto3";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

message  SensorMeteoData {
  int32 id = 1;
  message RawMeteoData {
    float temperature = 1;
    float humidity = 2;
    google.protobuf.Timestamp timestamp = 3;
  }
}

message SensorPollutionData {
  int32 id = 1;
  message RawPollutionData {
    float co2 = 1;
    google.protobuf.Timestamp timestamp = 2;
  }
}


service Server {
  rpc sendMeteoData (SensorMeteoData) returns (google.protobuf.Empty) {}
  rpc sendPollutionData (SensorPollutionData) returns (google.protobuf.Empty) {}
}